Update and organize the content on Deployments

shavidissa opened the pull request:
The commit:
- Organize the content.
- Make sure content adheres to the style guide.
- Fixes minor grammar issue.

shavidissa commented:
/assign @zacharysarah 
/assign @Bradamant3 
/assign @cody-clark 
/assign @zhangxiaoyu-zidif

shavidissa commented:
Please help review the content :)

sftim commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -69,84 +68,71 @@ In this example:
-To see the ReplicaSet (`rs`) created by the deployment, run `kubectl get rs`:
+  Follow the steps given below create the above Deployment:
+
+  Before you begin, make sure your Kubernetes cluster is up and running.
```
This reminds me of the `{{% capture prerequisites %}}` at the start of a Task page.

Maybe some of this page works better as a Task?

kbhawkey commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -17,10 +17,10 @@ weight: 30
 [ReplicaSets](/docs/concepts/workloads/controllers/replicaset/).
 
-You describe a _desired state_ in a Deployment object, and the Deployment controller changes the actual state to the desired state at a controlled rate. You can define Deployments to create new ReplicaSets, or to remove existing Deployments and adopt all their resources with new Deployments.
+You describe a _desired state_ in a Deployment, and the Deployment controller changes the actual state to the desired state at a controlled rate. You can define Deployments to create new ReplicaSets, or to remove existing Deployments and adopt all their resources with new Deployments.
```
I think in this first sentence, `... Deployment object,` helps indicate that Deployment is a resource, as well as a Controller,  and that you can define this resource in k8s configuration.

shavidissa replied:
The style guide mentions not to use  'Deployment object'.. instead to say the Deployment.  That's why I removed it. Is the update ok in that sense? 
[1] https://kubernetes.io/docs/contribute/style/style-guide/#use-camel-case-for-api-objects

kbhawkey commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -17,10 +17,10 @@ weight: 30
 
 {{< note >}}
-You should not manage ReplicaSets owned by a Deployment. All the use cases should be covered by manipulating the Deployment object. Consider opening an issue in the main Kubernetes repository if your use case is not covered below.
+You should not manage ReplicaSets owned by a Deployment. Consider opening an issue in the main Kubernetes repository if your use case is not covered below.
```
Could this note be further clarified? The sentence removed is confusing, but more information is needed.
Also, the note seems out of place. Should it moved under the [Use Case](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#use-case) section?

zacharysarah replied:
Let's avoid "should" when possible. Can we instead say something like:
> Do not manage ReplicaSets owned by a Deployment.

kbhawkey commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -54,12 +54,11 @@ In this example:
-  {{< /note >}}
+    {{< note >}}
+    The `matchLabels` is a map of {key,value} pairs. A single {key,value} in the `matchLabels` map
+    is equivalent to an element of `matchExpressions`, whose key field is "key", the operator is "In",
```
- You may want to add, The `matchLabels` field is a ...
- It may be helpful to link to the definition of `matchExpressions` here, otherwise the rest of the explanation seems out of context.
- whose `key` field is equal to "key", the `operator` field is set to "In", and the `values` list contains only one item, "value"?
What requirements are ANDed?

shavidissa replied:
+1. Thank you! I will update the content.

shavidissa replied:
I couldn't find a place to link it. The section was copied from [1]. Therefore, I made sure the content match.

[1] https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements

kbhawkey commented on content/en/docs/concepts/workloads/controllers/deployment.md:
````diff
@@ -166,151 +152,174 @@ A Deployment's rollout is triggered if and only if the Deployment's pod template
-nginx-deployment-1564180365-z9gth   1/1       Running   0          14s
-```
+* After the rollout succeeds, you may want to `get` the Deployment:
 
````
Instead of `get` the Deployment, 
... you can view the Deployment [by running `kubectl get deployments]

kbhawkey commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -166,151 +152,174 @@ A Deployment's rollout is triggered if and only if the Deployment's pod template
 ### Rollover (aka multiple updates in-flight)
 
-Each time a new deployment object is observed by the Deployment controller, a ReplicaSet is created to bring up
+Each time a new Deployment is observed by the Deployment controller, a ReplicaSet is created to bring up
```
Is the Deployment resource updated/created?

shavidissa replied:
Good catch. It was actually giving the wrong meaning. I reworded the content.

shavidissa commented:
@kbhawkey thanks a lot for the review! Good feedback :)
I updated the content.. can you take a look and give an _lgtm_ if it good to go?

steveperry-53 commented:
@shavidissa, Thanks for doing this! A couple of suggestions:

1) Make sure all instances of Deployment and Pod are capitalized.

2) Remove the phrase "running at all times". I think that's not accurate.

3) Remove "Open port 80 so that the container can send and accept traffic". That statement is incorrect. The Deployment manifest has nothing to do with opening a port. In fact, I'd suggest removing `containerPort` from the example manifest. It's an informational field that has no effect on anything, and people tend to misunderstand it.

kbhawkey commented:
hello @shavidissa. Just went through the page again.
I think this page needs to be broken down into several pages; I became lost on the single page, and some of the information seems incomplete. Perhaps another issue/PR is needed. 
Here are a few things I found.
- `Follow the steps given below create the above Deployment:`, fix.
- There are two places in the doc where a large section is repeated, `When you inspect the Deployment`. I don't think this explanation is needed x 2.
- Update a Deployment -- the relevant steps are not completely clear.
- Check wording under the `Rollover` section. Usage of `start` vs `starts`.
- Section, `Rolling Back a Deployment`, check `out put` vs. `output`.
- After paragraph that contains, `Press Ctrl-C`, there are code blocks that are not rendering correctly.
- There are two main headers that are similar, `Use Case` and `Use Cases`.

shavidissa commented:
Thanks a lot, @steveperry-53!  Good Catch.. I updated the content as per your feedback. I did not update the YAML file to remove the port 80 for now but I removed the explanation from the doc. I will check around a bit and update it in a subsequent PR. is that ok? 

> @shavidissa, Thanks for doing this! A couple of suggestions:
> 
> 1. Make sure all instances of Deployment and Pod are capitalized.
> 2. Remove the phrase "running at all times". I think that's not accurate.
> 3. Remove "Open port 80 so that the container can send and accept traffic". That statement is incorrect. The Deployment manifest has nothing to do with opening a port. In fact, I'd suggest removing `containerPort` from the example manifest. It's an informational field that has no effect on anything, and people tend to misunderstand it.

shavidissa commented:
Thanks a lot for the review @kbhawkey! 
> hello @shavidissa. Just went through the page again.
> I think this page needs to be broken down into several pages; I became lost on the single page, and some of the information seems incomplete. Perhaps another issue/PR is needed.
I agree.. Maybe the best way to go about is to create an issue and then address it in a subsequent PR?
> Here are a few things I found.
> 
> * `Follow the steps given below create the above Deployment:`, fix.
Good catch. Fixed
> * There are two places in the doc where a large section is repeated, `When you inspect the Deployment`. I don't think this explanation is needed x 2.
Removed the second instance.
> * Update a Deployment -- the relevant steps are not completely clear.
Did you find any specific area unclear? Trying to see where we can update the content further. Maybe we ca do it in a subsequent PR too after an issue is created?
> * Check wording under the `Rollover` section. Usage of `start` vs `starts`.
Fixed
> * Section, `Rolling Back a Deployment`, check `out put` vs. `output`.
Fixed
> * After paragraph that contains, `Press Ctrl-C`, there are code blocks that are not rendering correctly.
Good catch! Fixed
> * There are two main headers that are similar, `Use Case` and `Use Cases`.
Nice catch again! This has been coming down.. I updated it.

zacharysarah commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -54,99 +54,86 @@ In this example:
+    The `matchLabels` field is a map of {key,value} pairs. A single {key,value} in the `matchLabels` map
+    is equivalent to an element of `matchExpressions`, whose key field is "key" the operator is "In",
+    and the values array contains only "value".
+    All of the requirements, from both `matchLabels` and `matchExpressions` are ANDed together – they must all be
```
Grammar nits:
> All of the requirements, from both `matchLabels` and `matchExpressions`, must be satisfied in order to match.

zacharysarah commented on content/en/docs/concepts/workloads/controllers/deployment.md:
```diff
@@ -54,99 +54,86 @@ In this example:
+
+  Before you begin, make sure your Kubernetes cluster is up and running.
+
+  1. Create the Deployment, run the following command:
```
> 1. Create the Deployment by running the following command:

zacharysarah commented on content/en/docs/concepts/workloads/controllers/deployment.md:
````diff
@@ -54,99 +54,86 @@ In this example:
+    kubectl apply -f https://k8s.io/examples/controllers/nginx-deployment.yaml
+    ```
+
+  2. Next, let's run `kubectl get deployments` to check if the Deployment is created. If the Deployment is still being created, the output is similar to the following :
````
Per the style guide, [avoid using we](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we), plus grammar nits: 
> 2. Run `kubectl get deployments` to check if the Deployment was created. If the Deployment is still being created, the output is similar to the following:

shavidissa commented:
> @shavidissa Good work! ✨ Some grammar nits, and remember to avoid using "we".

Thank you for the review @zacharysarah! Updated the content. Noted on avoiding "we"/ "us" .

shavidissa commented:
Hi @zacharysarah,

Will you be able to help lgtm and approve the PR if the content is good to go, please? :)

zacharysarah commented:
/lgtm
/approve
